pkg/provider/handlers/read.go

```diff
@@ -47,6 +49,7 @@ func MakeReadHandler(client *containerd.Client) func(w http.ResponseWriter, r *h
 				Secrets:     fn.secrets,
 				EnvVars:     fn.envVars,
 				EnvProcess:  fn.envProcess,
+				Limits:      &types.FunctionResources{Memory: memory.ToDec().String()},
```

Comment:
> If you still have the environment running, can you test with Mi units?
> My example with figlet uses Mi and yours uses M.

I tested it just now, seems like it drops the unit of measure, as soon as it is not in DecimalSI.
If the `.ToDec()` is removed, it is only parsing the power-of-two equivalents.

Any idea how to handle both? Seems like the parser comes from the K8s API and I can't find any resource for that.

Reply:
What do we do in openfaas/faas-netes?

https://github.com/openfaas/faas-netes/blob/91bd12070db97f4bc5f2bdb7b0a5043f7d04c89e/pkg/k8s/function_status.go#L16

Does it help?

So given your input of `20 Mi`, what did containerd show?

```
sudo ctr -n openfaas-fn c ls
sudo ctr -n openfaas-fn c info figlet

sudo ctr -n openfaas-fn t metrics figlet
```

Reply:
```
sudo ctr -n openfaas-fn t metrics figlet
ID        TIMESTAMP                                  
figlet    2021-09-22 10:10:42.833830836 +0000 UTC    

METRIC                   VALUE         
memory.usage_in_bytes    1740800       
memory.limit_in_bytes    20971520      
memory.stat.cache        270336        
cpuacct.usage            11297941      
cpuacct.usage_percpu     [11297941]    
pids.current             6             
pids.limit               0     

```

20971520 / 1024 / 1024 = 20 Mb

Reply:
faas-netes using a different API to handle resources, so I don't think this is relevant in this case.

I tried to use the k8s API for this, like it is used in [`deploy.go`](https://github.com/openfaas/faasd/blob/master/pkg/provider/handlers/deploy.go#L151), to achive automatic conversion and precise unit of mesure.
Currently for this reasons I think we should stick to this and find a solution to be able to use this with the power-of-two equivalents.

Reply:
I am not sure where the requirement came from for supporting `M` and `G`.

What solutions can you think of and which do you recommend?

Reply:
The reason is, we already accept this units at deploy time.

The correct solution would be:
Continue accept `M` and `Mi` and if we got, for example `129M` we should display it as `123Mi`.
I'll play around with the k8s resource API and try to force a result that is suitable for our use case.

Reply:
```go
package main

import (
	"fmt"

	"k8s.io/apimachinery/pkg/api/resource"
)

func main() {
	memoryLimit := int64(20971520)
	memory := resource.NewQuantity(memoryLimit, resource.BinarySI)

	fmt.Println(memory.String())
	fmt.Println(memory.ToDec().String())
}
```

Result:

```bash
go run .
20Mi
20Mi
```